src/librustdoc/html/static/main.js

```diff
@@ -2843,22 +2839,14 @@ function defocusSearchBar() {
         }
     }
 
-    window.addSearchOptions = function(crates) {
+    function addSearchOptions(crates) {
         var elem = document.getElementById("crate-search");
 
         if (!elem) {
             enableSearchInput();
             return;
         }
-        var crates_text = [];
-        if (Object.keys(crates).length > 1) {
-            for (var crate in crates) {
-                if (hasOwnProperty(crates, crate)) {
-                    crates_text.push(crate);
-                }
-            }
-        }
-        crates_text.sort(function(a, b) {
+        crates.sort(function(a, b) {
```

Comment:
I just thought: now that the crate list is separated from the search index, it might be nice to sort them when generating the crate list in the rust code instead of doing it client side every time. What do you think? If you think it's too much, please add a FIXME comment so it can be taken care of later on.

Reply:
Yep, I'd be happy to do this!